### PR TITLE
gcpsecrets: fix new provider function

### PR DESCRIPTION
**Pull request by abursavich**
Resolves https://github.com/helmfile/vals/pull/101#discussion_r1096528265

### CODE REVIEW

File: pkg/providers/gcpsecrets/gcpsecrets.go [hunk 1]
```diff
@@ -15,42 +15,26 @@ import (
 
 // Format: ref+gcpsecrets://project/mykey[?version=VERSION][&fallback=value=valuewhenkeyisnotfound][&optional=true]#/yaml_or_json_key/in/secret
 type provider struct {
-	client   *sm.Client
-	ctx      context.Context
```
Comment (yxxhero) [inline pkg/providers/gcpsecrets/gcpsecrets.go:19]: why remove ctx filed?

Comment (abursavich) [reply pkg/providers/gcpsecrets/gcpsecrets.go:19]: It's not used anywhere.

Comment (yxxhero) [reply pkg/providers/gcpsecrets/gcpsecrets.go:19]: @mumoshu

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**